### PR TITLE
Geocoding

### PR DESCRIPTION
**Pull request by infntprimes**
Implement the ability to look up a latitude/longitude for a given zipcode via google's geocoding api, as well as cache the lookup for 30 days. 

`docs/geocoding.md` is the relevant design doc

### CODE REVIEW

File: airquality/generate_report.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+def create_report(request_body):
+    """
+    responsible for creating a report, called by our task queue's generate_report_handler
+    """
+    report_json = json.loads(request_body)
+    zipcode = report_json['zipcode']
+    start_date = report_json['start_date']
+    end_date = report_json['end_date']
+
+    latitude, longitude = get_latitude_longitude_for_zipcode(zipcode)
```
Comment (kmcmanus) [inline airquality/generate_report.py]: Do we want to return these?

File: airquality/tests.py [hunk 2]
```diff
@@ -268,3 +273,44 @@ def test_new_report_error_message_dates_out_of_range(self):
...
+        since each call is costing $$$ (but negligible for scope of project)
+        """
+        gm = googlemaps.Client(key=GOOGLE_API_KEY)
+        zipcode = '85259'
+
+        status, latlong = get_latitude_longitude_from_google_maps(zipcode)
+
+        self.assertTrue(status)
+        self.assertIsNotNone(latlong[0])
+        self.assertIsNotNone(latlong[1])
```
Comment (kmcmanus) [inline airquality/tests.py]: Can we test that _some known_ zip code returns the lat and long we want, or are within some distance to a lat and long we know?

File: geocoding/geocode.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    if cached_value is not None:
+        lat_long = cached_value.split("/")
+        latitude = lat_long[0]
+        longitude = lat_long[1]
+        return latitude, longitude
+    else:
+        success, lat_long = get_latitude_longitude_from_google_maps(zipcode)
+        if not success:
+            raise Exception('"DEBUG: could not retrieve latitude and longitude from gmaps API for zipcode {0} for reason: {1}'.format(zipcode,lat_long))
+        latitude = lat_long[0]
```
Comment (kmcmanus) [inline geocoding/geocode.py]: Can we do
```python
(latitide, longitude) = lat_long
```
?

File: geocoding/geocode.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
+from django_air_quality.privatesettings import GOOGLE_API_KEY
+from django.core.cache import cache
+from django_air_quality.settings import CACHE_TIMEOUT_SECONDS
+import googlemaps
+
+def get_latitude_longitude_for_zipcode(zipcode):
```
Comment (kmcmanus) [inline geocoding/geocode.py]: I'd love to see you rework this to use a decorator to handle the caching.

File: geocoding/geocode.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        longitude = lat_long[1]
+        cache.set(str(zipcode), str("{0}/{1}".format(latitude,longitude)), timeout=CACHE_TIMEOUT_SECONDS)
+        return latitude, longitude
+
+
+def get_latitude_longitude_from_google_maps(zipcode):
+    """
+    Attempts to extract a latitude and longitude from google's geocoding API for a provided zipcode
+    :param zipcode: zipcode to search for
+    :return: (true, [latitude,longitude]) if successful. Returns (false, error_message) if unsuccesful
```
Comment (kmcmanus) [inline geocoding/geocode.py]: I'm personally not a fan of this `flag, result` paradigm in python - what's the advantage over just raising an exception?

File: airquality/generate_report.py (hunk 1 above)
Comment (infntprimes) [reply airquality/generate_report.py]: not yet, this function will ultimately use these values to call other bigquery modules

**Pull request merged**